### PR TITLE
add warning message if serviceWorker is not available for some reason

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/browser/provider/index.ts [hunk 1]
```diff
@@ -240,6 +244,17 @@ export default class BrowserProvider {
         await browserTools.maximize(this._getWindowDescriptor(browserId));
     }
 
+    private async _ensureRetryTestPagesWarning (browserId: string): Promise<void> {
```
Comment (miherlosev) [inline src/browser/provider/index.ts:247]: I guess it should be a little bit more complicated:
* if connection is not secure, we should raise an error and abort test execution. Secure connection is a option usage requirement. 
Customer can fix this error by turning on the secure connection (specify 'localhost' as a 'host' parameter value or run TestCafe via HTTPS).

* if one of used browsers is not supported ServiceWorker, we should raise a warning. Customer cannot fix it. Also, it allows to run tests in such way: `testcafe chrome, ie tests --retry-test-pages`.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: src/browser/provider/index.ts (hunk 1 above)
Comment (AndreyBelym) [reply src/browser/provider/index.ts:247]: Agree with @miherlosev in general, but Safari will adds complexity here: it supports Service Workers only from version 12 and newer

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Comment by AlexKamaev** [comment]
FPR

File: test/server/runner-test.js [hunk 2]
```diff
@@ -350,6 +350,64 @@ describe('Runner', () => {
...
+                .src('test/server/data/test-suites/basic/testfile2.js')
+                .run()
+                .then(() => {
+                    throw new Error('Promise rejection expected');
+                })
+                .catch(err => {
+                    expect(err.message).eql(
+                        'Cannot enable the \'retryTestPages\' option. To enable the \'retryTestPages\' option please do one of the following:\n' +
+                        '-- specify the \'localhost\' value for the \'hostname\' option\n' +
+                        '-- run TestCafe over HTTPS protocol\n'
```
Comment (titerman) [inline test/server/runner-test.js:371]: I don't like this hammerhead (?) error message - we should probably fix it later

**Review by titerman — APPROVED** [review]

File: src/utils/is-localhost.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const LOCALHOST_NAMES = [
```
Comment (miherlosev) [inline src/utils/is-localhost.js:1]: Rewrite a new files with TypeScript.

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by AndreyBelym — APPROVED** [review]

File: test/server/runner-test.js (hunk 2 above)
Comment (titerman) [inline test/server/runner-test.js:371]:
```suggestion
'
                        'Cannot enable the \'retryTestPages\' option. Apply one of the following two solutions:\n' +
                        '-- set \'localhost\' as the value of the \'hostname\' option\n' +
                        '-- run TestCafe over HTTPS\n'
```

File: test/functional/fixtures/regression/gh-5239/test.js [hunk 4]
```diff
@@ -35,31 +45,52 @@ function createServer () {
...
+            return run({
+                retryTestPages: true,
+                browsers:       'ie',
+                src:            './testcafe-fixtures/warnings-test.js',
+                reporter:       customReporter
+            })
+                .then(() => {
+                    expect(warnings).eql([
+                        'The \'retryTestPages\' option was not enabled in the "ie" browser. ' +
+                        'Please ensure your browser supports Service Worker API (https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API)\n'
```
Comment (titerman) [inline test/functional/fixtures/regression/gh-5239/test.js:91]:
```suggestion
                        'Cannot enable the \'retryTestPages\' option in ie. ' +
                        'Please ensure that your version of ie supports the Service Worker API (https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).\n''
```

**Review by titerman — CHANGES_REQUESTED** [review]

**Review by miherlosev — APPROVED** [review]

**Comment by AlexKamaev** [comment]
@titerman FPR

**Review by titerman — APPROVED** [review]

**Pull request merged**